src/main/java/io/cdap/plugin/snowflake/sink/batch/SnowflakeBatchSink.java

```diff
@@ -74,13 +76,16 @@ public void prepareRun(BatchSinkContext context) {
     LineageRecorder lineageRecorder = new LineageRecorder(context, config.getReferenceName());
     lineageRecorder.createExternalDataset(inputSchema);
     // Record the field level WriteOperation
-    if (inputSchema.getFields() != null && !inputSchema.getFields().isEmpty()) {
+    if (inputSchema != null && inputSchema.getFields() != null && !inputSchema.getFields().isEmpty()) {
       String operationDescription = String.format("Wrote to Snowflake table '%s'", config.getTableName());
       lineageRecorder.recordWrite("Write", operationDescription,
                                   inputSchema.getFields().stream()
                                     .map(Schema.Field::getName)
                                     .collect(Collectors.toList()));
     }
+    // set error details provider
+    context.setErrorDetailsProvider(new ErrorDetailsProviderSpec(SnowflakeErrorDetailsProvider.class.getName()));
```

Comment:
This should be done before `context.addOutput(Output.of(config.getReferenceName(), new SnowflakeOutputFormatProvider(config)))`

Reply:
updated